src/libraries/System.Threading.Thread/tests/ThreadTests.cs

```diff
@@ -647,6 +647,19 @@ public static void NameTest()
             });
         }
 
+        [Fact]
+        public static void ThreadNameDoesNotAffectProcessName()
+        {
+            // On Linux, changing the main thread name affects ProcessName.
+            // To avoid that, .NET ignores requests to change the main thread name.
+            RemoteExecutor.Invoke(() =>
+            {
+                const string threadName = "my-thread";
+                Thread.CurrentThread.Name = threadName;
+                Assert.NotEqual(threadName, Process.GetCurrentProcess().ProcessName);
```

Comment:
```suggestion
                const string ThreadName = "my-thread";
                Thread.CurrentThread.Name = ThreadName;
                Assert.NotEqual(ThreadName, Process.GetCurrentProcess().ProcessName);
                Assert.Equal(ThreadName, Thread.CurrentThread.Name);
```

Reply:
Do we have tests that change a non-main thread name and validate it roundtrips?

Reply:
By rountrip you mean: _when we ask the OS the thread name after setting it_?
I had a look in the `ThreadTests`, and there doesn't seem to be such a test.

The Thread.Name is considered a write-once value. It is stored in a field which the getter returns. There is no code to get the name from the OS.

Reply:
> By rountrip you mean: when we ask the OS the thread name after setting it?

Yup.  Mind adding one?

Reply:
I meant literally just set the property and then get the property.

Reply:
ok. There is a `NameTest` that does it for a non-main thread.
I'll add it here to cover the main-thread case also.